core/file_walker.go

```diff
@@ -1,50 +1,70 @@
 package core
 
 import (
-	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 )
 
 type File struct {
 	Name  string
+	Dir   *File
```

Comment:
Renamed to `Parent`. I'll remove State.ancestors in a different MR.
Regarding space efficiency: We need to find other ways to reduce RAM usage.